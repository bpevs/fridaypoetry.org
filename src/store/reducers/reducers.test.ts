import { clone } from "lodash";
import reducer from "./reducers";


const defaultState = {
  poems: {
    allIds: undefined,
    byId: undefined
  }
};

test("Should have default state", () => {
  expect(reducer(undefined, { type: undefined })).toMatchSnapshot();
});

describe("FETCH_IDS_SUCCESS", function() {
  beforeEach(() => {
    this.prevState = clone(defaultState);
    this.action = { type: "FETCH_IDS_SUCCESS" };
  });

  test("Should update initial ids", () => {
    this.action.payload = {
      ids: [ "1-id", "2-id", "3-id" ]
    };

    expect(reducer(this.prevState, this.action)).toMatchSnapshot();
  });

  test("Should update subsequent ids", () => {
    this.prevState.poems.allIds = [ "1-id", "3-id", "4-id" ];
    this.action.payload = {
      ids: [ "2-id", "3-id", "5-id" ]
    };

    expect(reducer(this.prevState, this.action)).toMatchSnapshot();
  });
});

describe("FETCH_POEM_SUCCESS", function() {
  beforeEach(() => {
    this.prevState = clone(defaultState);
    this.action = { type: "FETCH_POEM_SUCCESS" };
  });

  test("FETCH_POEM_SUCCESS should update initial poem", () => {
    this.action.payload = {
      poem: {
        author: "poem-6-author",
        text: "poem-6-text",
        title: "poem-6-title"
      },
      poemId: "6-id"
    };

    expect(reducer(this.prevState, this.action)).toMatchSnapshot();
  });

  test("FETCH_POEM_SUCCESS should update subsequent poem", () => {
    this.prevState.poems.allIds = [ "1-id", "3-id", "4-id" ];
    this.prevState.poems.byId = {
      "3-id": {
        author: "poem-3-author",
        text: "poem-3-text",
        title: "poem-3-title"
      }
    };
    this.action.payload = {
      poem: {
        author: "poem-author",
        text: "poem-text",
        title: "poem-title"
      },
      poemId: "2-id"
    };

    expect(reducer(this.prevState, this.action)).toMatchSnapshot();
  });
});
