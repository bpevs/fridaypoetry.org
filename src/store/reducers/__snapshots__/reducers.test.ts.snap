// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should have default state 1`] = `
Object {
  "poems": Object {
    "allIds": Array [],
    "byId": Object {},
  },
}
`;

exports[`FETCH_IDS_SUCCESS Should update initial ids 1`] = `
Object {
  "poems": Object {
    "allIds": Array [
      "3-id",
      "2-id",
      "1-id",
    ],
    "byId": Object {},
  },
}
`;

exports[`FETCH_IDS_SUCCESS Should update subsequent ids 1`] = `
Object {
  "poems": Object {
    "allIds": Array [
      "5-id",
      "4-id",
      "3-id",
      "2-id",
      "1-id",
    ],
    "byId": Object {},
  },
}
`;

exports[`FETCH_POEM_SUCCESS FETCH_POEM_SUCCESS should update initial poem 1`] = `
Object {
  "poems": Object {
    "allIds": Array [
      "6-id",
      "4-id",
      "3-id",
      "1-id",
    ],
    "byId": Object {
      "6-id": Object {
        "author": "poem-6-author",
        "text": "poem-6-text",
        "title": "poem-6-title",
      },
    },
  },
}
`;

exports[`FETCH_POEM_SUCCESS FETCH_POEM_SUCCESS should update subsequent poem 1`] = `
Object {
  "poems": Object {
    "allIds": Array [
      "4-id",
      "3-id",
      "2-id",
      "1-id",
    ],
    "byId": Object {
      "2-id": Object {
        "author": "poem-author",
        "text": "poem-text",
        "title": "poem-title",
      },
      "3-id": Object {
        "author": "poem-3-author",
        "text": "poem-3-text",
        "title": "poem-3-title",
      },
    },
  },
}
`;
